litex/soc/doc/__init__.py

```diff
@@ -0,0 +1,329 @@
+# This file is Copyright (c) 2020 Sean Cross <sean@xobs.io>
+# License: BSD
+
+import os
+import pathlib
+import datetime
+
+from litex.soc.interconnect.csr import _CompoundCSR
+from .csr import DocumentedCSRRegion
+from .module import gather_submodules, ModuleNotDocumented, DocumentedModule, DocumentedInterrupts
+from .rst import reflow
+
+default_sphinx_configuration = """
+project = '{}'
+copyright = '{}, {}'
+author = '{}'
+extensions = [
+    'sphinx.ext.autosectionlabel',
+    'sphinxcontrib.wavedrom',{}
+]
+templates_path = ['_templates']
+exclude_patterns = []
+offline_skin_js_path = "https://wavedrom.com/skins/default.js"
+offline_wavedrom_js_path = "https://wavedrom.com/WaveDrom.js"
+html_theme = 'alabaster'
+html_static_path = ['_static']
+"""
+
+
+def sub_csr_bit_range(busword, csr, offset):
+    nwords = (csr.size + busword - 1)//busword
+    i = nwords - offset - 1
+    nbits = min(csr.size - i*busword, busword) - 1
+    name = (csr.name + str(i) if nwords > 1 else csr.name).upper()
+    origin = i*busword
+    return (origin, nbits, name)
+
+
+def print_svd_register(csr, csr_address, description, length, svd):
```

Comment:
The `svd` stuff should probably be in the same place we generate the json and c headers rather than in the doc?

Reply:
Is there an easy way in python to convert a print() to a string concatenation?  The existing litex stuff creates one big string inside the functions in `export.py`, then writes them all out at once.  lxsocdoc, meanwhile, takes a stream-based approach and generally writes to the files as it goes.

Reply:
Maybe something similar to what we are doing in the build backends:
````
from litex.build import tools
svd = []
svd.append("line1")
svd.append("line2")
tools.write_to_file("registers.svd", "\n".join(svd))
````

Reply:
As per your suggestion, it's now in `export.py`.

Reply:
Thanks!